### PR TITLE
unify SystemTime classes

### PR DESCRIPTION
**Pull request by teosarca**

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/test/java/org/compiere/util/TimeUtilTest.java [hunk 1]
```diff
@@ -622,10 +621,11 @@ public void test_isDateOrTimeClass()
...
 	{
-		SystemTime.setTimeSource(
-				FixedTimeSource.ofZonedDateTime(LocalDate.of(2020, Month.APRIL, 29)
-						.atTime(LocalTime.of(13, 14))
-						.atZone(ZoneId.of("UTC+5"))));
+		SystemTime.setFixedTimeSource("2020-04-29T13:14:00+05:00");
+		// SystemTime.setTimeSource(
+		// 		FixedTimeSource.ofZonedDateTime(LocalDate.of(2020, Month.APRIL, 29)
+		// 				.atTime(LocalTime.of(13, 14))
+		// 				.atZone(ZoneId.of("UTC+5"))));
```
Comment (metas-ts) [inline backend/de.metas.adempiere.adempiere/base/src/test/java/org/compiere/util/TimeUtilTest.java:628]:
```suggestion
```

File: backend/de.metas.util/src/main/java/de/metas/util/time/SystemTime.java [hunk 2]
```diff
@@ -1,59 +1,26 @@
...
-
-import lombok.NonNull;
-import lombok.experimental.UtilityClass;
-
 /**
- * Code taken from the book "Test Driven", Chapter 7 ("Test-driving the
- * unpredictable") by Lasse Koskela.
- *
- * @author ts
+ * Please consider {@link de.metas.common.util.time.SystemTime}.
```
Comment (metas-ts) [inline backend/de.metas.util/src/main/java/de/metas/util/time/SystemTime.java:16]:
```suggestion
 * @deprecated please consider {@link de.metas.common.util.time.SystemTime}.
```

File: misc/de-metas-common/de-metas-common-util/src/main/java/de/metas/common/util/time/SystemTime.java [hunk 3]
```diff
@@ -68,6 +73,11 @@ public void setFixedTimeSource(@NonNull final ZonedDateTime date)
 		setTimeSource(FixedTimeSource.ofZonedDateTime(date));
 	}
 
+	public void setFixedTimeSource(@NonNull final String zonedDateTime)
```
Comment (metas-ts) [inline misc/de-metas-common/de-metas-common-util/src/main/java/de/metas/common/util/time/SystemTime.java:76]: i suggest you add a javadoc with something like "see {@link ZonedDateTimeparse(String)}"

File: backend/de.metas.util/src/main/java/de/metas/util/time/TimeSource.java [hunk 4]
```diff
@@ -25,7 +25,7 @@
  */
 
 @FunctionalInterface
-public interface TimeSource
+public interface TimeSource extends de.metas.common.util.time.TimeSource
```
Comment (metas-ts) [inline backend/de.metas.util/src/main/java/de/metas/util/time/TimeSource.java:28]: Deprecate this timesource too?

File: backend/de.metas.util/src/main/java/de/metas/util/time/SystemTime.java (hunk 2 above)
Comment (teosarca) [reply backend/de.metas.util/src/main/java/de/metas/util/time/SystemTime.java:16]: yup... and additionally i will also replace the current usages. Just to not introduce another 600 warnings.

File: backend/de.metas.util/src/main/java/de/metas/util/time/TimeSource.java (hunk 4 above)
Comment (teosarca) [reply backend/de.metas.util/src/main/java/de/metas/util/time/TimeSource.java:28]: actually, I will delete it

File: misc/de-metas-common/de-metas-common-util/src/main/java/de/metas/common/util/time/SystemTime.java (hunk 3 above)
Comment (teosarca) [reply misc/de-metas-common/de-metas-common-util/src/main/java/de/metas/common/util/time/SystemTime.java]: done

File: backend/de.metas.util/src/main/java/de/metas/util/time/SystemTime.java (hunk 2 above)
Comment (teosarca) [reply backend/de.metas.util/src/main/java/de/metas/util/time/SystemTime.java]: update: i've removed that class

**Pull request merged**